File: Competitive Programming/Archive/old python/4.median-of-two-sorted-arrays.py
#
# @lc app=leetcode id=4 lang=python3
#
# [4] Median of Two Sorted Arrays
#

# @lc code=start
class Solution:
    def findMedianSortedArrays(self, nums1, nums2):
        arr = sorted(nums1 + nums2)
        mid = (0+(len(arr)-1))//2

        if len(arr) % 2 != 0:
            return arr[mid]
        return (arr[mid]+arr[mid+1])/2

# @lc code=end

File: Competitive Programming/_ archive/LC JavaScript Solutions/4.median-of-two-sorted-arrays.js
/*
 * @lc app=leetcode id=4 lang=javascript
 *
 * [4] Median of Two Sorted Arrays
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
  arr = nums1.concat(nums2).sort((a, b) => a - b);

  if (arr.length % 2 === 0) {
    return (
      (arr[Math.floor(arr.length / 2 - 1)] + arr[Math.floor(arr.length / 2)]) /
      2
    );
  }
  return arr[Math.floor(arr.length / 2)];
};
// @lc code=end
